src/components/asthma/components/AsthmaControlCalendar/AsthmaControlCalendar.tsx

```diff
@@ -0,0 +1,157 @@
+            setLogEntries(logEntries);
+        });
+    }, [], [props.previewState]);
+
+    const stateConfiguration: CalendarDayStateConfiguration = {
+        'not-determined': {
+            style: {
+                background: '#000',
+                color: '#fff'
+            },
+            streak: true,
+            streakColor: '#DBDBDB'
+        },
+        'not-controlled': {
+            style: {
+                background: '#F86A5C',
+                color: '#fff'
+            },
+            streak: true,
+            streakColor: '#FCCCC7'
+        },
+        'controlled': {
+            style: {
+                background: '#35A6A0',
+                color: '#fff'
+            },
+            streak: true,
+            streakColor: '#CCE9E7'
+        }
+    };
+
+    const computeStateForDay = (date: Date): string => {
+        let status = computeAsthmaControlState(logEntries, date).status;
+        if (status === 'not-determined') {
+            return 'not-determined';
+        } else if (status === 'not-controlled') {
+            return 'not-controlled';
+        } else if (status === 'controlled') {
+            return 'controlled';
+        } else if (isSameDay(date, new Date())) {
+            return 'today';
+        } else if (isAfter(date, new Date())) {
+            return 'future';
+        } else {
+            return 'no-data';
+        }
+    };
+
+    const onDayClicked = (date: Date): void => {
+        if (props.previewState || isAfter(date, new Date())) return;
+        MyDataHelps.openApplication(props.dayViewUrl + '?date=' + formatISO(date, {representation: 'date'}), {modal: true});
+    };
+
+    const renderDay = (year: number, month: number, day?: number): React.JSX.Element => {
+        return <CalendarDay
+            year={year}
+            month={month}
+            day={day}
+            stateConfiguration={stateConfiguration}
+            computeStateForDay={computeStateForDay}
+            onClick={onDayClicked}
+        />;
+    };
+
+    let intervalStart = dateRangeContext?.intervalStart ?? props.intervalStart ?? startOfMonth(new Date());
+
+    const getLogEntries = () => {
+        if (logEntries.length === 0) {
+            return null;
+        }
+
+        let today = startOfToday();
+        let logEntryLookup: Record<string, AsthmaLogEntry | undefined> = {};
+
+        let earliestLogEntry = [...logEntries].sort((e1, e2) => e1.identifier.localeCompare(e2.identifier))[0];
+        let earliestDate = parseISO(earliestLogEntry.identifier);
+        for (let i = 0; i <= differenceInDays(today, earliestDate); i++) {
+            let targetDate = add(earliestDate, {days: i});
+            if (!isBefore(targetDate, intervalStart) && isBefore(targetDate, add(intervalStart, {months: 1}))) {
+                logEntryLookup[dateToAsthmaLogEntryIdentifier(targetDate)] = logEntries.find(e => e.identifier === dateToAsthmaLogEntryIdentifier(targetDate));
```

Comment:
Remove calling dateToAsthmaLogEntryIdentifier(targetDate from each iteration in find

Reply:
Done.